contentcuration/contentcuration/tasks.py

```diff
@@ -165,3 +177,219 @@ def sendcustomemails_task(subject, message, query):
 @app.task(name="ensure_versioned_database_exists_task")
 def ensure_versioned_database_exists_task(channel_id, channel_version):
     ensure_versioned_database_exists(channel_id, channel_version)
+
+
+def _validate_audit_request(channel_id, user_id):
+    """Validate user and channel for audit request."""
+    try:
+        user = User.objects.get(pk=user_id)
+    except User.DoesNotExist:
+        logger.error(f"User with id {user_id} does not exist")
+        return None, None
+
+    try:
+        channel = Channel.objects.get(pk=channel_id)
+    except Channel.DoesNotExist:
+        logger.error(f"Channel with id {channel_id} does not exist")
+        return None, None
+
+    if not channel.editors.filter(pk=user_id).exists() and not user.is_admin:
+        logger.error(
+            f"User {user_id} is not an editor of channel {channel_id} and is not an admin"
+        )
+        return None, None
+
+    if not channel.main_tree.published:
+        logger.error(f"Channel {channel_id} is not published")
+        return None, None
+
+    return user, channel
+
+
+def _calculate_included_licenses(channel, published_data_version, channel_version):
+    """Calculate and cache included_licenses if not already present."""
+    included_licenses = published_data_version.get("included_licenses")
+    if not included_licenses:
+        published_nodes = (
+            channel.main_tree.get_descendants()
+            .filter(published=True)
+            .exclude(kind_id=content_kinds.TOPIC)
+        )
+        license_ids = list(
+            published_nodes.exclude(license=None)
+            .values_list("license", flat=True)
+            .distinct()
+        )
+        included_licenses = sorted(set(license_ids))
+        published_data_version["included_licenses"] = included_licenses
+        logger.info(
+            f"Calculated included_licenses for channel {channel.id} version {channel_version}: {included_licenses}"
+        )
+    return included_licenses
```

Comment:
To retrieve the included licenses, we should also use the unversioned database in storage, just as we do in the `_process_special_permissions_licenses` method. Then, to prevent us from having to download the channel database twice, we should do both operations within the same `with using_temp_migrated_content_database` context.